app/org/sagebionetworks/bridge/models/schedules/CompoundActivity.java

```diff
@@ -0,0 +1,118 @@
+package org.sagebionetworks.bridge.models.schedules;
+
+import java.util.List;
+import java.util.Objects;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.google.common.collect.ImmutableList;
+
+import org.sagebionetworks.bridge.BridgeUtils;
+
+/** Compound activity, which can include any number of schemas and surveys. */
+@JsonDeserialize(builder = CompoundActivity.Builder.class)
+public final class CompoundActivity {
+    private final List<SchemaReference> schemaList;
+    private final List<SurveyReference> surveyList;
+    private final String taskIdentifier;
+
+    /** Private constructor, use builder to construct. */
+    private CompoundActivity(List<SchemaReference> schemaList, List<SurveyReference> surveyList,
+            String taskIdentifier) {
+        this.schemaList = schemaList;
+        this.surveyList = surveyList;
+        this.taskIdentifier = taskIdentifier;
```

Comment:
Just a note that I don't think there would be a case where there was a surveyList AND a taskIdentifier (but if there's a taskIdentifier there should probably be at least one schedule in the list), but that's just something to check when validating I think.